shell/app/config-page/components/modal/modal.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import React from 'react';
+import { Modal } from 'core/nusi';
+import {
+  Info as IconInfo,
+  Attention as IconAttention,
+  CheckOne as IconCheckOne,
+  CloseOne as IconCloseOne,
+} from '@icon-park/react';
+import { map } from 'lodash';
+import { statusColorMap } from 'app/config-page/utils';
+import i18n from 'i18n';
+
+const widthMap = {
+  small: 350,
+  default: 500,
+  big: 650,
+  large: 800,
+};
+
+const stateIconMap = {
+  info: <IconInfo size="24" fill={statusColorMap.info} />,
+  success: <IconCheckOne size="24" fill={statusColorMap.success} />,
+  warning: <IconAttention size="24" fill={statusColorMap.warning} />,
+  danger: <IconCloseOne size="24" fill={statusColorMap.danger} />,
+};
+
+export default (props: CP_MODAL.Props) => {
+  const { props: configProps, children, footer, state, operations, execOperation } = props || {};
+  const [visible, setVisible] = React.useState(state?.visible ?? false);
+
+  React.useEffect(() => {
+    setVisible(state?.visible ?? false);
+  }, [state]);
+
+  const { size = 'default', status, title, content, ...rest } = configProps || {};
+
+  const onCancel = () => {
+    setVisible(false);
+    operations?.onCancel && execOperation(operations.onCancel);
+  };
+
+  const onOk = () => {
+    setVisible(false);
+    operations?.onOk && execOperation(operations.onOk);
+  };
+
+  const _title = (status || title) && (
```

Comment:
`_` prefix is unusually used in function temp variable,  this seems is not a temp variable

Reply:
That's ok.